### PR TITLE
feat(tflite): add private inference helpers and tests

### PR DESCRIPTION
**Pull request by Borda**
This pull request introduces a new TFLite inference helper module for RF-DETR exported models, along with a comprehensive test suite. The main focus is to provide lightweight, dependency-minimal utilities for running inference on TFLite models outside the training stack, and to ensure robust and reliable operation through detailed unit tests.

**Key additions and improvements:**

### TFLite Inference Utilities
* Added `src/rfdetr/export/_tflite/inference.py`, which provides helper functions for TFLite model inference:
  - [`_create_interpreter`](diffhunk://#diff-b97d0660e2c271d5846f16dd227daa7803091b2df11a2bd71cfaace54bd4c820R1-R140): Loads and prepares a TFLite interpreter, supporting both `tflite-runtime` and `tensorflow.lite` as backends, and prints input/output tensor shapes.
  - [`_run_inference`](diffhunk://#diff-b97d0660e2c271d5846f16dd227daa7803091b2df11a2bd71cfaace54bd4c820R1-R140): Handles image preprocessing (including normalization and resizing), runs inference, and decodes detection outputs into pixel-space bounding boxes using the `supervision` library. Supports both RGB and grayscale images and is robust to output tensor ordering.
  - Includes a numerically stable `_softmax` function for post-processing logits.

### Comprehensive Testing
* Added `tests/export/test_tflite_inference.py` with detailed unit tests for the new inference helpers:
  - Tests for interpreter creation, backend fallback logic, tensor allocation, and correct handling of both string and `Path` model paths.
  - Tests for inference correctness, including detection thresholding, pixel-space box scaling, handling of RGB and grayscale images, and robustness to output tensor reordering.
  - Uses extensive mocking to simulate TFLite interpreters and image inputs, ensuring tests do not require actual TFLite models or hardware.

These changes make it easy to deploy and test RF-DETR TFLite models in environments where only minimal dependencies are available, and ensure correctness and robustness through thorough testing.

### CODE REVIEW

File: src/rfdetr/export/_tflite/inference.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    environments).
+
+    Args:
+        model_path: Path to the ``.tflite`` model file.
+
+    Returns:
+        An allocated TFLite interpreter ready for inference.
+
+    Examples:
+        >>> interp = _create_interpreter("model_float32.tflite")  # doctest: +SKIP
```
Comment (Borda) [inline src/rfdetr/export/_tflite/inference.py]:
```suggestion
```

File: src/rfdetr/export/_tflite/inference.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        _Interpreter = _tflite.Interpreter  # noqa: N806
+    except ImportError:
+        try:
+            import tensorflow as _tf
+
+            _Interpreter = _tf.lite.Interpreter  # noqa: N806
+        except ImportError as exc:
+            raise ImportError(
+                "TFLite inference requires either 'tflite-runtime' or 'tensorflow'. "
+                "Install one: pip install tflite-runtime  OR  pip install tensorflow"
```
Comment (Borda) [inline src/rfdetr/export/_tflite/inference.py]:
```suggestion
                "Install one: `pip install tflite-runtime`  OR  `pip install tensorflow`"
```

File: src/rfdetr/export/_tflite/inference.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+        threshold: Confidence threshold; detections below this are discarded.
+
+    Returns:
+        A tuple of ``(detections, pil_img)`` where ``detections`` contains
+        pixel-space ``xyxy`` boxes and ``pil_img`` is the original PIL image
+        at its original resolution.
+
+    Examples:
+        >>> interp = _create_interpreter("model.tflite")  # doctest: +SKIP
+        >>> dets, img = _run_inference(interp, "image.jpg", threshold=0.3)  # doctest: +SKIP
```
Comment (Borda) [inline src/rfdetr/export/_tflite/inference.py]:
```suggestion
```
